### ISSUE TITLE
The **nullable** and **defaultValue** and **value** attributes do not work together as one

### ISSUE THREAD

**Issue by batizhao** [liquibase/liquibase#2652]
## Environment

**Liquibase Version**: 4.6.2-4.8.0
**Liquibase Integration & Version**:  Spring Boot 2.6.4
**Database Vendor & Version**: MySQL 8.0.20
**Operating System Type & Version**: macOS 12.3

## Description
After upgrading 4.6.1->4.6.2+, Using the createTable tag, The **nullable** and **defaultValue** attributes do not work.

## databaseChangeLog
```xml
<changeSet author="tom" id="20220124-003">
    <createTable tableName="menu">
        <column autoIncrement="true" name="id" type="BIGINT" remarks="pk">
            <constraints nullable="false" primaryKey="true"/>
        </column>
        <column name="path" type="VARCHAR(255)" remarks="path" defaultValue="">
            <constraints nullable="false"/>
        </column>
    </createTable>
</changeSet>
```

## Liquibase 4.6.1 creates a correct table
```sql
`path` varchar(255) NOT NULL DEFAULT '' COMMENT 'path',
```

## Liquibase 4.6.2+ creates a incorrect table
```sql
`path` varchar(255) DEFAULT NULL COMMENT 'path',
```

**Comment by kdebski85** [comment]
I tested it on DB2 LUW 11.1 and Liquibase 4.9.0.
It worked as expected. Column 'path' is not null and has default.
This is probably db-specific issue, or it is fixed in 4.9.0.

**Comment by molivasdat** [comment]
Hi @batizhao Thanks for the alert about this error you are seeing.  Thanks for the feedback @kdebski85 .  Based on the comments by @kdebski85, this may be a database specific issue for MySQL.  @batizhao or you can try 4.9.0 and let us know your feedback.

**Comment by molivasdat** [comment]
Hi @batizhao 
Here are the results of my tests with changelog you provided and MYSQL 8.0.20
4.6.1
```
--  Changeset gh2652.xml::20220124-003::tom
CREATE TABLE new_db.menu (id BIGINT AUTO_INCREMENT NOT NULL, `path` VARCHAR(255) DEFAULT '' NOT NULL COMMENT 'path', CONSTRAINT PK_MENU PRIMARY KEY (id));

ALTER TABLE new_db.menu MODIFY COLUMN id BIGINT COMMENT 'pk';

ALTER TABLE new_db.menu MODIFY COLUMN `path` VARCHAR(255) COMMENT 'path';
```

4.6.2
```
--  Changeset gh2652.xml::20220124-003::tom
CREATE TABLE new_db.menu (id BIGINT AUTO_INCREMENT NOT NULL, `path` VARCHAR(255) DEFAULT '' NOT NULL COMMENT 'path', CONSTRAINT PK_MENU PRIMARY KEY (id));

ALTER TABLE new_db.menu MODIFY COLUMN id BIGINT COMMENT 'pk';

ALTER TABLE new_db.menu MODIFY COLUMN `path` VARCHAR(255) COMMENT 'path';
```

4.9.0
```
--  Changeset gh2652.xml::20220124-003::tom
CREATE TABLE new_db.menu (id BIGINT AUTO_INCREMENT NOT NULL, `path` VARCHAR(255) DEFAULT '' NOT NULL COMMENT 'path', CONSTRAINT PK_MENU PRIMARY KEY (id));

ALTER TABLE new_db.menu MODIFY COLUMN id BIGINT COMMENT 'pk';

ALTER TABLE new_db.menu MODIFY COLUMN `path` VARCHAR(255) COMMENT 'path';
```
They all look correct and are they same as what was requested.
What is different in your environment?

**Comment by erolatreston** [comment]
Please see https://github.com/liquibase/liquibase/issues/2657. The problem exists only for adding non-nullable columns when the `value` is also specified. If the value is not specified, default value is correctly set.

**Comment by batizhao** [comment]
The problem still exists on 4.9.0.
@molivasdat 

```xml
<dependency>
    <groupId>org.liquibase</groupId>
    <artifactId>liquibase-core</artifactId>
    <version>4.9.0</version>
</dependency>
```

**Comment by gdouyang** [comment]
#2563 
createTable where generator "CREATE TABLE" and "ALTER TABLE"，when exec "ALTER TABLE" the "nullabel" and "defaultValue" will be clear, the sql of `ALTER TABLE `stalber-test`.menu MODIFY COLUMN `path` VARCHAR(255) COMMENT 'path'` change to `ALTER TABLE `stalber-test`.menu MODIFY COLUMN `path` VARCHAR(255) default 'value' not null COMMENT 'path'` can fix it

**Comment by nvoxland** [comment]
This is working for me as of 4.13.0. I think it was addressed by #2752
